resources/pre-upgrade.sh

```diff
@@ -33,17 +33,36 @@ function dumpData(){
       local USERS
       USERS="$(mysql -uroot mysql -e "select GROUP_CONCAT(User) FROM user WHERE NOT User LIKE '%mysql.%' AND NOT User='root';" | tail -n +2 | sed 's/,/ /g')"
 
-      local user
-      for user in ${USERS}
-      do
-          local CREATE
-          CREATE="$(mysql --skip-column-names -A mysql -e "SET @@SESSION.print_identified_with_as_hex = 1; SHOW CREATE USER '${user}'")"
-          echo "${CREATE};" >> /alldb.sql
-      done
+      # as users may exists within the dump but need to be recreated with all privileges the user must be dropped first
+      # otherwise the creation would result in an ERROR 1396 (HY000) - see https://stackoverflow.com/a/6332971
+      if [[ "${USERS}" != "NULL" ]]; then
+        local user
+        for user in ${USERS}
+        do
+            echo "DROP user IF EXISTS '${user}';" >> /alldb.sql
```

Comment:
Ah shoot, I just see this now in the second reading. Currently we write a file into container file system. This file will 1. be lost and 2. enlarge the container file system if the container crashes so that someone might fall for the idea to re-create the container.

We should write the file into a volume (a volume that must exist in the old dogu version) so it may out-live the container.

Also there are several occurrences of this file name throughout this routine, we should extract the path into a variable.